### ISSUE TITLE
Build error on Fedora 36

### ISSUE THREAD

**Issue by devrimgunduz** [pgexporter/pgexporter#35]
Hi @jesperpedersen ,

I'm getting the following error while building 0.2.2 on Fedora 36 (OpenSSL 3.0.2). I can see that you fixed this on git head. Any chance to release a new version so that I can package pgexporter on F36?

Thanks! Devrim

> [ 45%] Building C object src/CMakeFiles/pgexporter.dir/libpgexporter/remote.c.o
[ 50%] Building C object src/CMakeFiles/pgexporter.dir/libpgexporter/security.c.o
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c: In function 'generate_md5':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:780:4: warning: 'MD5_Init' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  780 |    MD5_Init(&c);
      |    ^~~~~~~~
In file included from /var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:52:
/usr/include/openssl/md5.h:49:27: note: declared here
   49 | OSSL_DEPRECATEDIN_3_0 int MD5_Init(MD5_CTX *c);
      |                           ^~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:781:4: warning: 'MD5_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  781 |    MD5_Update(&c, str, length);
      |    ^~~~~~~~~~
/usr/include/openssl/md5.h:50:27: note: declared here
   50 | OSSL_DEPRECATEDIN_3_0 int MD5_Update(MD5_CTX *c, const void *data, size_t len);
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:782:4: warning: 'MD5_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  782 |    MD5_Final(digest, &c);
      |    ^~~~~~~~~
/usr/include/openssl/md5.h:51:27: note: declared here
   51 | OSSL_DEPRECATEDIN_3_0 int MD5_Final(unsigned char *md, MD5_CTX *c);
      |                           ^~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c: In function 'client_proof':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2101:4: warning: 'HMAC_CTX_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2101 |    HMAC_CTX* ctx = HMAC_CTX_new();
      |    ^~~~~~~~
In file included from /usr/include/openssl/ssl.h:37,
                 from /var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/include/pgexporter.h:42,
                 from /var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:30:
/usr/include/openssl/hmac.h:33:33: note: declared here
   33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
      |                                 ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2131:4: warning: 'HMAC_Init_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2131 |    if (HMAC_Init_ex(ctx, s_k, s_k_length, EVP_sha256(), NULL) != 1)
      |    ^~
/usr/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2136:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2136 |    if (HMAC_Update(ctx, (unsigned char*)client_first_message_bare, client_first_message_bare_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2141:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2141 |    if (HMAC_Update(ctx, (unsigned char*)",", 1) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2146:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2146 |    if (HMAC_Update(ctx, (unsigned char*)server_first_message, server_first_message_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2151:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2151 |    if (HMAC_Update(ctx, (unsigned char*)",", 1) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2156:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2156 |    if (HMAC_Update(ctx, (unsigned char*)client_final_message_wo_proof, client_final_message_wo_proof_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2161:4: warning: 'HMAC_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2161 |    if (HMAC_Final(ctx, c_s, &length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2176:4: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2176 |    HMAC_CTX_free(ctx);
      |    ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2196:7: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2196 |       HMAC_CTX_free(ctx);
      |       ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c: In function 'salted_password':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2221:4: warning: 'HMAC_CTX_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2221 |    HMAC_CTX* ctx = HMAC_CTX_new();
      |    ^~~~~~~~
/usr/include/openssl/hmac.h:33:33: note: declared here
   33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
      |                                 ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2249:4: warning: 'HMAC_Init_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2249 |    if (HMAC_Init_ex(ctx, password, password_length, EVP_sha256(), NULL) != 1)
      |    ^~
/usr/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2254:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2254 |    if (HMAC_Update(ctx, (unsigned char*)salt, salt_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2259:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2259 |    if (HMAC_Update(ctx, (unsigned char*)&one, sizeof(one)) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2264:4: warning: 'HMAC_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2264 |    if (HMAC_Final(ctx, &Ui_prev[0], &Ui_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2273:7: warning: 'HMAC_CTX_reset' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2273 |       if (HMAC_CTX_reset(ctx) != 1)
      |       ^~
/usr/include/openssl/hmac.h:34:27: note: declared here
   34 | OSSL_DEPRECATEDIN_3_0 int HMAC_CTX_reset(HMAC_CTX *ctx);
      |                           ^~~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2279:7: warning: 'HMAC_Init_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2279 |       if (HMAC_Init_ex(ctx, password, password_length, EVP_sha256(), NULL) != 1)
      |       ^~
/usr/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2284:7: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2284 |       if (HMAC_Update(ctx, &Ui_prev[0], size) != 1)
      |       ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2289:7: warning: 'HMAC_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2289 |       if (HMAC_Final(ctx, &Ui[0], &Ui_length) != 1)
      |       ^~
/usr/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2305:4: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2305 |    HMAC_CTX_free(ctx);
      |    ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2317:7: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2317 |       HMAC_CTX_free(ctx);
      |       ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c: In function 'salted_password_key':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2336:4: warning: 'HMAC_CTX_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2336 |    HMAC_CTX* ctx = HMAC_CTX_new();
      |    ^~~~~~~~
/usr/include/openssl/hmac.h:33:33: note: declared here
   33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
      |                                 ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2353:4: warning: 'HMAC_Init_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2353 |    if (HMAC_Init_ex(ctx, salted_password, salted_password_length, EVP_sha256(), NULL) != 1)
      |    ^~
/usr/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2358:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2358 |    if (HMAC_Update(ctx, (unsigned char*)key, strlen(key)) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2363:4: warning: 'HMAC_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2363 |    if (HMAC_Final(ctx, r, &length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2372:4: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2372 |    HMAC_CTX_free(ctx);
      |    ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2384:7: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2384 |       HMAC_CTX_free(ctx);
      |       ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c: In function 'server_signature':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2518:4: warning: 'HMAC_CTX_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2518 |    HMAC_CTX* ctx = HMAC_CTX_new();
      |    ^~~~~~~~
/usr/include/openssl/hmac.h:33:33: note: declared here
   33 | OSSL_DEPRECATEDIN_3_0 HMAC_CTX *HMAC_CTX_new(void);
      |                                 ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2554:4: warning: 'HMAC_Init_ex' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2554 |    if (HMAC_Init_ex(ctx, s_k, s_k_length, EVP_sha256(), NULL) != 1)
      |    ^~
/usr/include/openssl/hmac.h:43:27: note: declared here
   43 | OSSL_DEPRECATEDIN_3_0 int HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len,
      |                           ^~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2559:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2559 |    if (HMAC_Update(ctx, (unsigned char*)client_first_message_bare, client_first_message_bare_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2564:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2564 |    if (HMAC_Update(ctx, (unsigned char*)",", 1) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2569:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2569 |    if (HMAC_Update(ctx, (unsigned char*)server_first_message, server_first_message_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2574:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2574 |    if (HMAC_Update(ctx, (unsigned char*)",", 1) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2579:4: warning: 'HMAC_Update' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2579 |    if (HMAC_Update(ctx, (unsigned char*)client_final_message_wo_proof, client_final_message_wo_proof_length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:45:27: note: declared here
   45 | OSSL_DEPRECATEDIN_3_0 int HMAC_Update(HMAC_CTX *ctx, const unsigned char *data,
      |                           ^~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2584:4: warning: 'HMAC_Final' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2584 |    if (HMAC_Final(ctx, r, &length) != 1)
      |    ^~
/usr/include/openssl/hmac.h:47:27: note: declared here
   47 | OSSL_DEPRECATEDIN_3_0 int HMAC_Final(HMAC_CTX *ctx, unsigned char *md,
      |                           ^~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2593:4: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2593 |    HMAC_CTX_free(ctx);
      |    ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/security.c:2614:7: warning: 'HMAC_CTX_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
 2614 |       HMAC_CTX_free(ctx);
      |       ^~~~~~~~~~~~~
/usr/include/openssl/hmac.h:35:28: note: declared here
   35 | OSSL_DEPRECATEDIN_3_0 void HMAC_CTX_free(HMAC_CTX *ctx);
      |                            ^~~~~~~~~~~~~
[ 55%] Building C object src/CMakeFiles/pgexporter.dir/libpgexporter/server.c.o
[ 60%] Building C object src/CMakeFiles/pgexporter.dir/libpgexporter/shmem.c.o
[ 65%] Building C object src/CMakeFiles/pgexporter.dir/libpgexporter/utils.c.o
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/utils.c: In function 'pgexporter_read_version':
/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/src/libpgexporter/utils.c:1756:4: warning: ignoring return value of 'fgets' declared with attribute 'warn_unused_result' [-Wunused-result]
 1756 |    fgets(&buf[0], sizeof(buf), file);
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 70%] Linking C shared library libpgexporter.so
/usr/bin/ld: /tmp/ccbvwYbe.ltrans0.ltrans.o: warning: relocation against `stdout@@GLIBC_2.2.5' in read-only section `.text'
/usr/bin/ld: /tmp/ccbvwYbe.ltrans0.ltrans.o: relocation R_X86_64_PC32 against symbol `shmem' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[3]: *** [src/CMakeFiles/pgexporter.dir/build.make:293: src/libpgexporter.so.0.2.2] Error 1
make[3]: Leaving directory '/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/build'
make[2]: *** [CMakeFiles/Makefile2:148: src/CMakeFiles/pgexporter.dir/all] Error 2
make[2]: Leaving directory '/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/build'
make[1]: *** [Makefile:156: all] Error 2
make[1]: Leaving directory '/var/lib/pgsql/rpmcommon/BUILD/pgexporter-0.2.2/build'
error: Bad exit status from /var/tmp/rpm-tmp.DgsluM (%build)

**Comment by jesperpedersen** [comment]
(Likewise with pgagroal and pgmoneta)

**Comment by jesperpedersen** [comment]
Can you share

```
make VERBOSE=1
```

?

**Comment by devrimgunduz** [comment]
There is no Makefile here. Am I missing aything?

**Comment by jesperpedersen** [comment]
The `cmake` command will create one, here

```
https://git.postgresql.org/gitweb/?p=pgrpms.git;a=blob;f=rpm/redhat/main/common/pgexporter/main/pgexporter.spec;h=9c13c9d5f65a215cb6ef7c5c6fcd81aeaa988a69;hb=HEAD#l23
```

**Comment by devrimgunduz** [comment]
Hi Jesper,

Here is the output:

> [  5%] Linking C shared library libpgexporter.so
> cd /home/devrim/belgeler/Devrim/Projects/repo/pgrpms/rpm/redhat/main/common/pgexporter/main/pgexporter-0.2.2/build/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/pgexporter.dir/link.txt --verbose=1
> /usr/lib64/ccache/gcc -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  	 -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wformat -Wformat-security -fstack-protector-strong -fPIC -fPIE -O3 -DNDEBUG -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -Wl,--build-id=sha1 -Wl,-dT,/home/devrim/belgeler/Devrim/Projects/repo/pgrpms/rpm/redhat/main/common/pgexporter/main/pgexporter-0.2.2/.package_note-pgexporter-0.2.2-1.f42.x86_64.ld -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -shared -Wl,-soname,libpgexporter.so.0 -o libpgexporter.so.0.2.2 CMakeFiles/pgexporter.dir/libpgexporter/configuration.c.o CMakeFiles/pgexporter.dir/libpgexporter/logging.c.o CMakeFiles/pgexporter.dir/libpgexporter/management.c.o CMakeFiles/pgexporter.dir/libpgexporter/memory.c.o CMakeFiles/pgexporter.dir/libpgexporter/message.c.o CMakeFiles/pgexporter.dir/libpgexporter/network.c.o CMakeFiles/pgexporter.dir/libpgexporter/prometheus.c.o CMakeFiles/pgexporter.dir/libpgexporter/queries.c.o CMakeFiles/pgexporter.dir/libpgexporter/remote.c.o CMakeFiles/pgexporter.dir/libpgexporter/security.c.o CMakeFiles/pgexporter.dir/libpgexporter/server.c.o CMakeFiles/pgexporter.dir/libpgexporter/shmem.c.o CMakeFiles/pgexporter.dir/libpgexporter/utils.c.o  /usr/lib64/libev.so /usr/lib64/libssl.so /usr/lib64/libcrypto.so /usr/lib64/libsystemd.so 
> /usr/bin/ld: /tmp/cc1xX63s.ltrans0.ltrans.o: warning: relocation against `stdout@@GLIBC_2.2.5' in read-only section `.text'
> /usr/bin/ld: /tmp/cc1xX63s.ltrans0.ltrans.o: relocation R_X86_64_PC32 against symbol `shmem' can not be used when making a shared object; recompile with -fPIC
> /usr/bin/ld: final link failed: bad value
> collect2: error: ld returned 1 exit status
> make[2]: *** [src/CMakeFiles/pgexporter.dir/build.make:293: src/libpgexporter.so.0.2.2] Error 1
>

**Comment by jesperpedersen** [comment]
Hmm, ok. 

As you can see `-fPIC` is there... but there are additional switches than upstream, so maybe one of those is causing the issue... I assume that the `.c` compile look similar

**Comment by jesperpedersen** [comment]
Ok, have it replicated locally -- only happens with RPM based build.

Also, clang doesn't work for some reason

**Comment by jesperpedersen** [comment]
It is related to `-fPIE`...

**Comment by jesperpedersen** [comment]
Fixed - thanks for the report !

The functionality will be available in the next patch release (`0.2.3`)
